test/junit/scala/reflect/internal/TypesTest.scala

```diff
@@ -58,4 +58,45 @@ class TypesTest {
         Assert.fail(xs.mkString("\n"))
     }
   }
+
+  @Test
+  def testSameTypesLub(): Unit = {
+    def testSameType(tpe: Type, num: Int = 5) = assert(lub(List.fill(num)(tpe)) =:= tpe)
+
+    testSameType(IntTpe)
+    testSameType(StringTpe)
+    testSameType(typeOf[Class[String]])
+    testSameType(LiteralType(Constant(1)))
+    testSameType(LiteralType(Constant("test")))
+  }
+
+  @Test
+  def testTypesLub(): Unit = {
+    val interestingCombos: Map[Type, List[List[Type]]] = Map(
+      IntTpe -> List(List(LiteralType(Constant(0)), LiteralType(Constant(1))),
+                     List(LiteralType(Constant(0)), IntTpe),
+                     List(ConstantType(Constant(0)), IntTpe),
+                     List(ConstantType(Constant(0)), LiteralType(Constant(1))),
+                     List(ConstantType(Constant(1)), LiteralType(Constant(1))), // Should this be LiteralType(Constant(1))?
```

Comment:
I remember not completely understanding why `LUB` of these two is `Int` and not `LiteralType(ConstantType(1))` or `ConstantType(ConstantType(1))`.